### PR TITLE
PIM-7891: update exports when an attribute is deleted

### PR DESCRIPTION
**Pull request by rouliane**
**Description (for Contributor and Core Developer)**
When an attribute is deleted in the PIM, product exports are now updated is the attribute was used as a filter

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/EventSubscriber/RemoveAttributeFiltersInJobInstancesOnAttributeDeletion.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            StorageEvents::POST_REMOVE => 'removeDeletedAttributeFromJobInstancesFilters',
+        ];
+    }
+
+    public function removeDeletedAttributeFromJobInstancesFilters(GenericEvent $event)
```
Comment (jmleroux) [inline src/Pim/Bundle/CatalogBundle/EventSubscriber/RemoveAttributeFiltersInJobInstancesOnAttributeDeletion.php]:
```suggestion
    public function removeDeletedAttributeFromJobInstancesFilters(GenericEvent $event): void
```

File: src/Pim/Bundle/CatalogBundle/EventSubscriber/RemoveAttributeFiltersInJobInstancesOnAttributeDeletion.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Akeneo PIM Enterprise Edition.
```
Comment (mmetayer) [inline src/Pim/Bundle/CatalogBundle/EventSubscriber/RemoveAttributeFiltersInJobInstancesOnAttributeDeletion.php]: nope ;)

File: src/Pim/Bundle/CatalogBundle/tests/integration/EventSubscriber/RemoveAttributeFiltersInJobInstancesOnAttributeDeletionIntegration.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+use Akeneo\Component\Batch\Model\JobInstance;
+use Akeneo\Test\Integration\TestCase;
+use Pim\Bundle\CatalogBundle\Entity\Attribute;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+class RemoveAttributeFiltersInJobInstancesOnAttributeDeletionIntegration extends TestCase
+{
+    public function testUpdateExportsFiltersOnAttributeDeletion()
+    {
+        $jobInstance = $this->createJobInstanceWithAttributeFilter('job1', ['a_yes_no', 'a_text', 'a_date']);
```
Comment (mmetayer) [inline src/Pim/Bundle/CatalogBundle/tests/integration/EventSubscriber/RemoveAttributeFiltersInJobInstancesOnAttributeDeletionIntegration.php]: I'd add a second job instance in the test

File: src/Pim/Bundle/CatalogBundle/EventSubscriber/RemoveAttributeFiltersInJobInstancesOnAttributeDeletion.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+use Akeneo\Component\Batch\Model\JobInstance;
+use Akeneo\Component\StorageUtils\Saver\BulkSaverInterface;
+use Akeneo\Component\StorageUtils\StorageEvents;
+use Doctrine\Common\Persistence\ObjectRepository;
+use Pim\Component\Catalog\Model\AttributeInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+class RemoveAttributeFiltersInJobInstancesOnAttributeDeletion implements EventSubscriberInterface
```
Comment (mmetayer) [inline src/Pim/Bundle/CatalogBundle/EventSubscriber/RemoveAttributeFiltersInJobInstancesOnAttributeDeletion.php]: add copyright and license tags

**Pull request merged**